src/dask_awkward/layers/layers.py

```diff
@@ -107,20 +107,30 @@ def mock_empty(self, backend: BackendT = "cpu") -> AwkwardArray:
         )
 
 
+def io_func_rer_wrapped(func: ImplementsIOFunction) -> bool:
+    return hasattr(func, "__rer_wrapped__")
+
+
+def maybe_unwrap(func: Callable) -> Callable:
+    if io_func_rer_wrapped(func):
+        return func.__rer_wrapped__  # type: ignore
+    return func
+
+
 def io_func_implements_projection(func: ImplementsIOFunction) -> bool:
-    return hasattr(func, "prepare_for_projection")
+    return hasattr(maybe_unwrap(func), "prepare_for_projection")
```

Comment:
I'd be tempted to remove the unwrapping here, and require that the caller unwrap the IO function. That way it's a bit more explicit?